include/swift/Syntax/SyntaxFactory.h

```diff
@@ -474,12 +474,20 @@ struct SyntaxFactory {
   /// Make a bare "()" void tuple type
   static TupleTypeSyntax makeVoidTupleType();
 
+  /// Make a tuple type from an array of types, with one space between each
+  /// type.
+  static TupleTypeSyntax
+  makeTupleType(llvm::ArrayRef<TupleTypeElementSyntax> Types);
```

Comment:
I think we'll need paren arguments here to, so folks can customize leading and trailing trivia on the left and right paren, respectively.